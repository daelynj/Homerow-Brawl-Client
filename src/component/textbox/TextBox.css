div.Textbox {
  display: grid;
  width: 540px;
  grid-template-columns: 120px 120px 120px;
  align-items: start;
  justify-content: space-between;
}
