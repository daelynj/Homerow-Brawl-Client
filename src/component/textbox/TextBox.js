import React from 'react';
import Text from './Text';

function TextBox(props) {
  let currentLetters = props.text[props.currentWord].slice(0, props.input.length);
  function renderText() {
    return (
      <Text
        text = {props.text}
        currentLetters = {currentLetters}
        input = {props.input}
      />
    )
  }

  return (
    <div className="TextBox">
      {renderText()}
    </div>
  )
}

export default TextBox;
