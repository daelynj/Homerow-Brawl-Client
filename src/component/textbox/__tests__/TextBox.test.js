import React from 'react';
import TextBox from '../TextBox';
import renderer from 'react-test-renderer';

describe("TextBox", () => {
  it("renders the expected text", () => {
    const tree = renderer.create(
      <TextBox
        text = {["This", "is", "text"]}
        textToCompare = {""}
        input = {""}
        currentWord = {0}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
