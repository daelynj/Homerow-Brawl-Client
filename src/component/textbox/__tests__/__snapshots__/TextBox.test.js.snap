// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox renders the expected text 1`] = `
<div
  className="Text"
>
  <div
    className="Words"
  >
    <span
      className="Word"
    >
      <span
        className="black"
      >
        T
      </span>
      <span
        className="black"
      >
        h
      </span>
      <span
        className="black"
      >
        i
      </span>
      <span
        className="black"
      >
        s
      </span>
    </span>
    <span
      className="Word"
    >
      <span
        className="black"
      >
        i
      </span>
      <span
        className="black"
      >
        s
      </span>
    </span>
    <span
      className="Word"
    >
      <span
        className="black"
      >
        t
      </span>
      <span
        className="black"
      >
        e
      </span>
      <span
        className="black"
      >
        x
      </span>
      <span
        className="black"
      >
        t
      </span>
    </span>
  </div>
</div>
`;
